### DISCUSSION TITLE
Disable  4-digit code  activation.

### DISCUSSION THREAD

**Discussion by waybarrios** [plausible/analytics#700] [Self-Hosted Support] [answered]
I am running local with docker-compose and I set DISABLE_AUTH and DISABLE_REGISTRATION. But I always get: 

```
Activate your account
Please enter the 4-digit code we sent to hello@local
```
Any suggestions to disable it? 

Thanks
Wayner

**Comment by ukutaht** [comment, marked as answer]
Hmm.

The app tries to verify each email address by sending a code, including the admin address. I suppose we could allow a mode where email confirmation is not required.

To fix your situation you can either:
1. Set up your mail server so you can activate the admin email address
2. Access the postgres database directly and `UPDATE users set email_verified=true;`. Instructions for accessing postgres directly: https://github.com/plausible/analytics/discussions/621

**Reply by martinord** [reply]
Thanks for your reply!

I've updated the status of email verification as you say, but it does not seem to make effect

**Reply by worosom** [reply]
Make sure to terminate your SQL command with a semicolon.
`docker exec -ti hosting_plausible_db_1 psql -U postgres plausible_db`
`UPDATE users set email_verified=true;`
After running these commands I was able to log in without having to enter the confirmation code.

**Reply by martinord** [reply]
You're right, missed the semicolon ;D What a noob error! Thanks a lot

**Comment by ianaya89** [comment]
I am also having issues with this. I have a self hosted instance in EC2 and when I try to activate the account the email is not being sent.
This is the log from the smpt container

```
Berkeley DB: Berkeley DB 5.3.28: (September  9, 2013)
Support for: crypteq iconv() IPv6 GnuTLS move_frozen_messages DKIM DNSSEC Event OCSP PRDR SOCKS TCP_Fast_Open
Lookups (built-in): lsearch wildlsearch nwildlsearch iplsearch cdb dbm dbmjz dbmnz dnsdb dsearch nis nis0 passwd
Authenticators: cram_md5 plaintext
Routers: accept dnslookup ipliteral manualroute queryprogram redirect
Transports: appendfile/maildir/mailstore autoreply lmtp pipe smtp
Fixed never_users: 0
Configure owner: 0:0
Size of off_t: 8
  244 delivering 1l9qJL-00003v-Kf
  244 R: dnslookup for ignacio.anaya89@gmail.com
  245 T: remote_smtp for ignacio.anaya89@gmail.com
  245 Connecting to gmail-smtp-in.l.google.com [2607:f8b0:400d:c08::1b]:25 ... failed: Cannot assign requested address
  245 LOG: MAIN
  245   H=gmail-smtp-in.l.google.com [2607:f8b0:400d:c08::1b] Cannot assign requested address
  245 Connecting to gmail-smtp-in.l.google.com [172.217.222.26]:25 ... failed: Connection timed out (timeout=5m)
  245 LOG: MAIN
  245   H=gmail-smtp-in.l.google.com [172.217.222.26] Connection timed out
  245 Connecting to alt1.gmail-smtp-in.l.google.com [2800:3f0:4003:c00::1a]:25 ... failed: Cannot assign requested address
  245 LOG: MAIN
  245   H=alt1.gmail-smtp-in.l.google.com [2800:3f0:4003:c00::1a] Cannot assign requested address
  245 Connecting to alt4.gmail-smtp-in.l.google.com [172.217.218.26]:25 ... failed: Connection timed out (timeout=5m)
  245 LOG: MAIN
  245   H=alt4.gmail-smtp-in.l.google.com [172.217.218.26] Connection timed out
  244 LOG: MAIN
  244   == ignacio.anaya89@gmail.com R=dnslookup T=remote_smtp defer (110): Connection timed out
  252 LOG: MAIN
  ```

**Reply by teito-dev** [reply]
It look's like gmail is blocking your email from being sent ,

Can you show a copy of your plausible-conf.env ? (with credentials removed i.e)

Unless you have a mailserver setup on that EC2 instance , 
you need to either setup a mail relay agent ,
or pass in the correct ,
SMTP_HOST_ADDR
SMTP_HOST_PORT
SMTP_USER_NAME
SMTP_USER_PWD

**Comment by nuno-agostinho** [comment]
Although a bit "dirty", I am using the following workaround to automatically disable email verification.

The `postgres` Docker image has a [`/docker-entrypoint.initdb.d/`](https://hub.docker.com/_/postgres) that automatically runs all `.sh` files inside when starting the service. So I have created a `init.sh` file with the following command:

```bash
#!/bin/sh
sleep 24 && psql -U 'postgres' -d 'plausible_db' -c "UPDATE users set email_verified=true;" &
```

Then I modified the `docker-compose.yml` at the `plausible-db` service to mount the `init.sh` file inside the `/docker-entrypoint.initdb.d` directory with read-only access (`ro`).

```
plausible_db:
    image: postgres:12
    restart: always
    volumes:
      - db-data:/var/lib/postgresql/data
      - ./init.sh:/docker-entrypoint-initdb.d/init.sh:ro
    environment:
      - POSTGRES_PASSWORD=postgres
```

In the end, this will run sometime after the admin user is automatically created, thus the only command I need to run is `docker-compose up -d` and wait ~30 seconds before using plausible.

@ukutaht I would love a cleaner way to remove email verification, such as using an environment variable. Also, the `DISABLE_AUTH` parameter maybe should disable email verification by default. BTW, I love plausible, thank you a lot for this great program! ❤️

**Reply by ukutaht** [reply]
Thanks @nuno-agostinho! I have a list of stuff for 1.4.0 release, adding this one as well.
